src/statistical/unigram.jl

```diff
@@ -0,0 +1,221 @@
+ WordTokenizers.Nodes("gu", -21.9203f0, 3074, 17, 18)
+ WordTokenizers.Nodes("gua", -23.776f0, 15259, 17, 19)
+ WordTokenizers.Nodes("ag", -34.1531f0, 3303, 19, 20)
+ WordTokenizers.Nodes("language", -11.1965f0, 7021, 14, 21)
+``` 
+"""
+function decode_forward(sp::SentencePieceModel, text::String)
+    results = Array{Nodes, 1}(undef, length(text))
+    scores = fill(-Inf, length(text))
+    scores[1] =0
+    for char_end in 1:length(text)
+        for char_start in 1:char_end
+            if text[char_start:char_end] in sp.vocab
+                subtokenid = getindex(sp, text[char_start:char_end])[1]
+                local_score = scores[char_start] + sp.logprob[subtokenid]
+                if local_score > scores[char_end]   
+                    results[char_end] = Nodes(text[char_start:char_end], local_score, subtokenid, char_start, char_end)
+                    scores[char_end] = local_score
+                end
+            end
+        end
+        if scores[char_end] == -Inf
+            results[char_end] = Nodes(text[char_end-1:char_end], -Inf, 1, char_end-1, char_end)
+            scores[char_end] = 0
+        end
+        if scores[char_end] == 0
+            results[char_end] = Nodes(text[char_end:char_end], -Inf, 1, char_end, char_end)
+        end
+    end
+    return(results)
+end
+
+"""
+    decode_backward(sp::SentencePieceModel,text::String)
+inputs nodes (i.e. output of `decode_forward`) and
+Return output of backword pass as mentioned [here](https://tejasvaidhyadev.github.io/blog/Sentencepiece), as an Array{String,1}
+# Example
+'''julia-repl
+julia> WordTokenizers.decode_backward(spm ,node)
+8-element Array{Any,1}:
+ WordTokenizers.Nodes("language", -11.1965f0, 7021, 14, 21)
+ WordTokenizers.Nodes(" ", -Inf32, 1, 13, 13)
+ WordTokenizers.Nodes("ulia", -22.49547f0, 19590, 9, 12)
+ WordTokenizers.Nodes("j", -Inf32, 1, 8, 8)
+ WordTokenizers.Nodes(" ", -Inf32, 1, 7, 7)
+ WordTokenizers.Nodes("love", -10.7128f0, 4584, 3, 6)
+ WordTokenizers.Nodes(" ", -Inf32, 1, 2, 2)
+ WordTokenizers.Nodes("I", -Inf32, 1, 1, 1)
+'''
+"""
+function decode_backward(sp::SentencePieceModel, nodes)
+    next_nodes = nodes[end]
+    best_seq = []
+    
+    while next_nodes.start > 1
+        node_value = next_nodes
+        next_nodes = nodes[(node_value.start)-1]
+        push!(best_seq, node_value)
+    end
+    push!(best_seq, next_nodes)
+    return(best_seq)
+end
+
+"""
+    tokenizer(sp::SentencePieceModel,text::AbstractString)
+It does all the preprocessing step needed and perform `decode_forward` and `decode_backward`
+ouput tokenize tokens as Array{String,1}
+"""
+function tokenizer(sp::SentencePieceModel, text::AbstractString)
+    tks = []
+    text = replace(text, " " => "_")
+    if text[1] != '_'
+        text = "_" * text
+    end
+    output = decode_forward(sp, text)
+    tokens = decode_backward(sp, output)
+    tokens = reverse(tokens)
+    for node in tokens
+        push!(tks, node.text)
+    end
+    tks = string.(tks)
```

Comment:
using a list comprehension means that this will end up concretely typed.
So won't need to call `string.(...)` on it.
It also means that can delete `tks` from start
```suggestion
    tks = [node.text for node in tokens]
```

Reply:
I will take care of all from next time